### PR TITLE
Switch to stylelint.

### LINKED ISSUE #23383: Switch to stylelint.

### PR DESCRIPTION
**Pull request by XhmikosR**
Fixes #23383 

Note to all: feel free to push to this branch, and merge from v4-dev, but please do not rebase until we are ready to merge to v4-dev.

TODO:

- [x] Make sure tests are green
- [x] Check for properties order
- [x] Tighten rules
- [x] Make sure Hound works

### CODE REVIEW

**Comment by wojtask9** [comment]
Probably you also want to use `stylelint-order` plugin (https://github.com/hudochenkov/stylelint-order).

**Comment by XhmikosR** [comment]
Yup, it's on my todo, but I thought I'd let the other team members chime in before I make a lot more changes myself :)

**Comment by andresgalante** [comment]
It'd be great if we make style lint format the code or use another formater. 
Fornater + lint = happiness 

Love this change 👍🏻

**Comment by XhmikosR** [comment]
I believe this is in a mergeable state; we can tighten the rules later more.

My only issue is freaking Hound. I don't know if it will stop complaining after this has been merged or not.

**Comment by bardiharborow** [comment]
@XhmikosR bit of a shot in the dark, but you may need to pass `scss: enabled: false` in `.hound.yml` to disable the default linter (which is now malfunctioning because you've removed the config file). My recollection is that Hound configurations are somewhat unintuitive in that respect.

**Comment by wolfy1339** [comment]
Hound uses the config of the parent branch, not the one in the current branch. The only way to test hound is to PR against this branch, in order to use the config in this branch

**Comment by XhmikosR** [comment]
OK, then we can ignore Hound in this PR.

@mdo: please have a look when you have some time and merge or let me know if you want me to make any changes. We'll need to tweak the rules more after the merge anyway; I have some of the rules set to `null` so that things pass currently.

**Comment by XhmikosR** [comment]
(I can't override Hound's blocking review, only repo owners can)

File: package.json [hunk 1]
```diff
@@ -117,9 +122,7 @@
     "build",
     "js/.eslintrc.json",
     "js/**/*.js",
-    ".scss-lint.yml",
-    "scss/**/*.scss",
-    "LICENSE"
```
Comment (mdo) [inline package.json]: Shouldn't the license stay?

**Comment by XhmikosR** [comment]
It's default included by npm.

File: package.json (hunk 1 above)
Comment (XhmikosR) [reply package.json]: It's default included in npm.

**Comment by XhmikosR** [comment]
@mdo: is there anything else holding us for this? Otherwise pull the trigger 🔫

**Pull request merged**